Получить всех пользователей, которым понравился пост или комментарий 1.0v

salikhdev opened the pull request.

salikhdev committed:
Получить всех пользователей, которым понравился пост или комментарий 1.0v

# Conflicts:
#	src/main/java/faang/school/postservice/controller/LikeController.java
#	src/main/java/faang/school/postservice/repository/LikeRepository.java
#	src/main/java/faang/school/postservice/service/LikeService.java
#	src/test/java/faang/school/postservice/service/LikeServiceTest.java

az3l1t commented on src/main/java/faang/school/postservice/mapper/UserMapper.java:
```diff
@@ -0,0 +1,9 @@
+
+@Mapper(componentModel = "spring")
+public interface UserMapper {
+
```
Зачем это тут?
бесполезный маппер

az3l1t commented on src/main/java/faang/school/postservice/service/LikeService.java:
```diff
@@ -0,0 +1,57 @@
+                .map(Like::getUserId)
+                .collect(Collectors.toList());
+
+        try {
```
повторяющийся код в двух методах, вынести в отдельный метод

az3l1t commented on src/main/java/faang/school/postservice/service/LikeService.java:
```diff
@@ -0,0 +1,57 @@
+                .map(Like::getUserId)
+                .collect(Collectors.toList());
+
+        try {
```
повторяющийся код в двух методах, вынести в отдельный метод

az3l1t commented on src/test/java/faang/school/postservice/ApplicationContextTest.java:
```diff
@@ -19,8 +19,8 @@ public class ApplicationContextTest {
     }
 
-    @Test
+  /*  @Test
```
комменты лучше не оставляй. если есть тест, который не должен работать, либо убирай, либо аннотацию @Disabled

az3l1t commented on src/main/java/faang/school/postservice/controller/LikeController.java:
```diff
@@ -0,0 +1,31 @@
+    private final LikeService likeService;
+
+    @GetMapping("/post/{postId}")
+    public List<UserDto> getUsersLikedToPost(@PathVariable long postId) {
```
входные данные никак не проверяются. а если null?
смысла дальше не будет логику исполнять

fladx replied:
> входные данные никак не проверяются. а если null? смысла дальше не будет логику исполнять

он примитив получает long, поэтому это уже как встроенная валидация

az3l1t replied:
а да, не заметил

az3l1t commented on src/main/java/faang/school/postservice/service/LikeService.java:
```diff
@@ -0,0 +1,57 @@
+    @Transactional(readOnly = true)
+    public List<UserDto> getLikedUsersToComment(long commentId) {
+        if (!commentService.existsById(commentId)) {
+            throw new IllegalArgumentException("Comment with id %s does not exist".formatted(commentId));
```
делаем свои excpetions, дальше будет легче обрабатывать

az3l1t commented on src/main/java/faang/school/postservice/service/LikeService.java:
```diff
@@ -0,0 +1,57 @@
+
+@Service
+@RequiredArgsConstructor
+public class LikeService {
```
добавь логирование: info, debug, error

fladx commented on src/main/java/faang/school/postservice/repository/CommentRepository.java:
```diff
@@ -10,4 +10,7 @@ public interface CommentRepository extends CrudRepository<Comment, Long> {
     @Query("SELECT c FROM Comment c WHERE c.post.id = :postId")
     List<Comment> findAllByPostId(long postId);
+
+    boolean existsById(long id);
```
это уже встроенный метод в Jpa

SnurnicynKonstantin replied:
Надо проверить, если встроенный реашет ту же задачу, то этот метод надо удалить

SnurnicynKonstantin commented on src/main/java/faang/school/postservice/exceptions/UserServiceConnectException.java:
```diff
@@ -0,0 +1,7 @@
+package faang.school.postservice.exceptions;
+
+public class UserServiceConnectException extends RuntimeException {
```
Нужно включить это исключение в handler - https://github.com/CorporationX/user_service/blob/chimera-master-stream8/src/main/java/school/faang/user_service/controller/handler/ControllerExceptionHandler.java

SnurnicynKonstantin commented on src/main/java/faang/school/postservice/mapper/UserMapper.java:
```diff
@@ -0,0 +1,9 @@
+import org.mapstruct.Mapper;
+
+@Mapper(componentModel = "spring")
+public interface UserMapper {
```
Обрати внимание, маппер пустой. Следует удалить

SnurnicynKonstantin reviewed (changes requested):
Все аккуратно, супер. 
Обрати на внимание на мои комментарии и комментарии ребят. Они все валидны

salikhdev committed:
Получить всех пользователей, которым понравился пост или комментарий 2.0v

salikhdev committed:
Получить всех пользователей, которым понравился пост или комментарий 2.1v

SnurnicynKonstantin commented on src/main/java/faang/school/postservice/repository/CommentRepository.java:
```diff
@@ -10,4 +10,7 @@ public interface CommentRepository extends CrudRepository<Comment, Long> {
     @Query("SELECT c FROM Comment c WHERE c.post.id = :postId")
     List<Comment> findAllByPostId(long postId);
+
+    boolean existsById(long id);
```
Этот метод уже есть в репозитории. Руками его создавать не нужно

SnurnicynKonstantin commented on src/main/java/faang/school/postservice/dto/Message.java:
```diff
@@ -0,0 +1,9 @@
+package faang.school.postservice.dto;
+
+
+public record Message(
```
Лучше переименовать в ErrorResponse. По текущему имени не понятно о чем это ДТО

SnurnicynKonstantin reviewed (changes requested):
Совсем мелоч осталась.

salikhdev committed:
Получить всех пользователей, которым понравился пост или комментарий 2.2v

# Conflicts:
#	src/main/java/faang/school/postservice/controller/GlobalExceptionHandler.java

salikhdev committed:
Получить всех пользователей, которым понравился пост или комментарий 2.4v

fladx commented on src/main/java/faang/school/postservice/repository/LikeRepository.java:
```diff
@@ -0,0 +1,15 @@
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public interface LikeRepository extends JpaRepository<Like, Long> {
```
Где аннотация репозитория?

SnurnicynKonstantin replied:
Нет аннотации Repository

fladx commented on src/test/java/faang/school/postservice/ApplicationContextTest.java:
```diff
@@ -20,6 +21,7 @@ public class ApplicationContextTest {
     }
 
     @Test
+    @Disabled
     void contextLoads() {
 
     }
```
почему выключил?

salikhdev replied:
Не нужно же

fladx commented on src/test/java/faang/school/postservice/service/LikeServiceTest.java:
```diff
@@ -0,0 +1,142 @@
+
+    private Stream<Like> likeStream;
+
+    @BeforeEach
+    public void setUp() {
+        likeStream = Stream.of(
+                Like.builder().id(1).userId(1L).build(),
+                Like.builder().id(2).userId(2L).build(),
+                Like.builder().id(3).userId(3L).build()
+        );
+
+        userList = List.of(
+                new UserDto(1L, "user1", "user1@gmail.com"),
+                new UserDto(2L, "user2", "user2@gmail.com"),
+                new UserDto(3L, "user3", "user3@gmail.com")
+        );
+    }
```
Мне кажется тут лучше испольхзовать @BeforeAll

fladx commented on src/test/java/faang/school/postservice/service/LikeServiceTest.java:
```diff
@@ -0,0 +1,142 @@
+
+    @Test
+    @DisplayName("Test getLikedUsersToPost Success")
+    public void testGetLikedUsersToPostSuccess() {
```
наименование тестов не соответствует документу команды

SnurnicynKonstantin reviewed (changes requested):
Посмотри замечания Влада

salikhdev committed:
Получить всех пользователей, которым понравился пост или комментарий 2.5v

SnurnicynKonstantin commented on src/test/java/faang/school/postservice/service/LikeServiceTest.java:
```diff
@@ -0,0 +1,142 @@
+
+    private Stream<Like> likeStream;
+
+    @BeforeEach
```
Можно @BeforeAll т.к. эти объекты создаются один раз

SnurnicynKonstantin commented on src/test/java/faang/school/postservice/ApplicationContextTest.java:
```diff
@@ -20,6 +21,7 @@ public class ApplicationContextTest {
     }
 
     @Test
+    @Disabled
```
Если пометил как disabled, может он не нужен вовсе? Если не нужен, стоит удалить

SnurnicynKonstantin reviewed (changes requested):
Предыдущие комментарии не исправлены

salikhdev committed:
Получить всех пользователей, которым понравился пост или комментарий 2.6v

SnurnicynKonstantin reviewed (changes requested):
Осталось конфликты решить

salikhdev committed:
Получить всех пользователей, которым понравился пост или комментарий 2.7v

SnurnicynKonstantin reviewed (approved):
Наконец-то доталкали задачку)
